src/marqo/core/index_management/index_management.py

```diff
@@ -450,14 +457,35 @@ def _add_default_query_profile(self, app: str) -> bool:
         if not os.path.exists(profile_path):
             logger.debug('Default query profile does not exist. Creating it')
 
-            query_profile = self._DEFAULT_QUERY_PROFILE_TEMPLATE
+            query_profile = self._DEFAULT_QUERY_PROFILE_TEMPLATE.format(
+                max_hits=self._max_search_limit,
+                max_offset=self._max_search_offset
+            )
             os.makedirs(os.path.dirname(profile_path), exist_ok=True)
             with open(profile_path, 'w') as f:
                 f.write(query_profile)
 
             return True
         return False
 
+    def _update_query_profile(self, app: str) -> bool:
+        profile_path = os.path.join(app, 'search/query-profiles', 'default.xml')
+        new_query_profile = self._DEFAULT_QUERY_PROFILE_TEMPLATE.format(
+            max_hits=self._max_search_limit,
+            max_offset=self._max_search_offset
+        )
+
+        if os.path.exists(profile_path):
+            with open(profile_path, 'r') as f:
+                existing_content = f.read()
+            if existing_content == new_query_profile:
+                return False
```

Comment:
shouldn't we have an elif branch here to actually write to the profile path file if it does not match the new query profile?

Reply:
I think the following block covers this

```os.makedirs(os.path.dirname(profile_path), exist_ok=True)
        with open(profile_path, 'w') as f:
            f.write(new_query_profile)```